execution_engine_testing/tests/src/test/contract_api/subcall.rs

```diff
@@ -154,17 +154,13 @@ fn should_add_all_gas_for_subcall() {
     let add_zero_gas_via_subcall_cost = builder.exec_costs(2)[0];
     let add_some_gas_via_subcall_cost = builder.exec_costs(3)[0];
 
-    assert!(add_some_gas_from_session_cost.value() > gas_to_add);
-    assert_eq!(
-        add_some_gas_from_session_cost.value(),
-        gas_to_add + add_zero_gas_from_session_cost.value()
-    );
-
-    assert!(add_some_gas_via_subcall_cost.value() > gas_to_add);
-    assert_eq!(
-        add_some_gas_via_subcall_cost.value(),
-        gas_to_add + add_zero_gas_via_subcall_cost.value()
+    let expected_gas = U512::from(StorageCosts::default().gas_per_byte()) * gas_to_add;
+    assert!(
+        add_zero_gas_from_session_cost.value() < add_zero_gas_via_subcall_cost.value(),
+        "subcall expected to cost more gas due to storing contract"
     );
+    assert!(add_some_gas_from_session_cost.value() > expected_gas);
+    assert!(add_some_gas_via_subcall_cost.value() > expected_gas);
```

Comment:
These changes look like the cost of execution changed - does it mean that some contracts' execution will change now?

Reply:
The intention here was to make a wasm cost "at least the expected_gas amount" as this PR disables import of internal 'gas' function. With `expected_gas` large enough, it should be a precise measurement. Previously, you could precisely add the amount of gas to the counter with the imported gas function.